faker/providers/misc/__init__.py

```diff
@@ -91,7 +92,8 @@ def binary(cls, length=(1 * 1024 * 1024)):
 
         Default blob size is 1 Mb.
         """
-        return os.urandom(length)
+        blob = [random.randrange(256) for o in range(length)]
+        return bytes(blob) if sys.version_info[0] >= 3 else bytearray(blob)
```

Comment:
not really. Python 2 bytes is an alias to str. So if you get 1 byte it will generate something like '[87]' (length 4 str).

Example:
===Python 3====
>>> byteArr = [23,67,75,254]
>>> bytes(byteArr)
b'\x17CK\xfe'
>>> len(bytes(byteArr))
4

===Python 2====
>>> byteArr = [23,67,75,254]
>>> bytes(byteArr)
'[23, 67, 75, 254]'
>>> len(bytes(byteArr))
17

>>> bytearray(byteArr)
bytearray(b'\x17CK\xfe')
>>> len(bytearray(byteArr))
4